10월 18, 2014 7:25:31 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter <init>
정보: Setting Up identifiers & ports for the network service to listen on
10월 18, 2014 7:25:32 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$StartHandler onNext
정보: StartTime: com.microsoft.wake.time.event.StartTime[1413627932138]
10월 18, 2014 7:25:32 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$AllocatedEvaluatorHandler onNext
정보: Received an AllocatedEvaluator. Submitting it.
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 1, Outstanding requests: 1
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 2, Outstanding requests: 1
10월 18, 2014 7:25:32 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$AllocatedEvaluatorHandler onNext
정보: Received an AllocatedEvaluator. Submitting it.
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 3, Outstanding requests: 1
10월 18, 2014 7:25:32 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$AllocatedEvaluatorHandler onNext
정보: Received an AllocatedEvaluator. Submitting it.
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 4, Outstanding requests: 0
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 4, Outstanding requests: 0
10월 18, 2014 7:25:32 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$AllocatedEvaluatorHandler onNext
정보: Received an AllocatedEvaluator. Submitting it.
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-1-1413627932150"
state: RUNNING
 
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-2-1413627932285"
state: RUNNING
 
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-4-1413627932379"
state: RUNNING
 
10월 18, 2014 7:25:32 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-3-1413627932346"
state: RUNNING
 
10월 18, 2014 7:25:35 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$ActiveContextHandler onNext
정보: Received a RunningEvaluator with ID: SGD_Linear
10월 18, 2014 7:25:35 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$ActiveContextHandler onNext
정보: Received a RunningEvaluator with ID: SGD_Linear
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$ActiveContextHandler onNext
정보: Received a RunningEvaluator with ID: SGD_Linear
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$ActiveContextHandler onNext
정보: Received a RunningEvaluator with ID: SGD_Linear
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: All context are running
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Setting Up Structures for creating Group Comm Operator Configurations
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Registering ComputeTask1 with jw-PC:7001
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Registering ComputeTask2 with jw-PC:7002
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Registering ComputeTask3 with jw-PC:7003
10월 18, 2014 7:25:36 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Creating Operator Configs
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$RunningTaskHandler onNext
정보: Task "ComputeTask3" is running!
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$RunningTaskHandler onNext
정보: Task "ComputeTask1" is running!
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$RunningTaskHandler onNext
정보: Task "ComputeTask2" is running!
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$RunningTaskHandler onNext
정보: Task "ControllerTask" is running!
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Task ComputeTask1 is done.
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Task ComputeTask3 is done.
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Task ComputeTask2 is done.
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Releasing Context SGD_Linear.
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Releasing Context SGD_Linear.
10월 18, 2014 7:25:37 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Releasing Context SGD_Linear.
10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onError
경고: Shutting down the Driver with an exception: 
java.lang.RuntimeException: No handler bound for FailedTask: com.microsoft.reef.driver.task.FailedTask id=ControllerTask failed: Task.call() threw an Exception.
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultTaskFailureHandler.onNext(DefaultTaskFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultTaskFailureHandler.onNext(DefaultTaskFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.runtime.common.evaluator.task.exceptions.TaskCallFailure: Task.call() threw an Exception.
	at com.microsoft.reef.runtime.common.evaluator.task.TaskRuntime.runTask(TaskRuntime.java:267)
	at com.microsoft.reef.runtime.common.evaluator.task.TaskRuntime.run(TaskRuntime.java:123)
	... 1 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: 6
	at com.microsoft.reef.examples.groupcomm.matmul.DenseVector.get(DenseVector.java:42)
	at snu.BDCS.SGD_Linear_Regression.VectorConcat.apply(VectorConcat.java:36)
	at snu.BDCS.SGD_Linear_Regression.VectorConcat.apply(VectorConcat.java:20)
	at com.microsoft.reef.io.network.group.impl.operators.basic.ReduceOp$Receiver.reduce(ReduceOp.java:118)
	at com.microsoft.reef.io.network.group.impl.operators.basic.ReduceOp$Receiver.reduce(ReduceOp.java:106)
	at snu.BDCS.SGD_Linear_Regression.ControllerTask.call(ControllerTask.java:94)
	at com.microsoft.reef.runtime.common.evaluator.task.TaskRuntime.runTask(TaskRuntime.java:262)
	... 2 more

10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Dirty shutdown with outstanding containers.
10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Force shutdown of: ProcessContainer{containedID='Node-3-1413627932346', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-3-1413627932346}
10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-3-1413627932346', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-3-1413627932346}
10월 18, 2014 7:25:37 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0xfeec80d2, /169.254.245.8:2203 => /169.254.245.8:16656] local: /169.254.245.8:16656 remote: /169.254.245.8:2203
10월 18, 2014 7:25:37 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:37 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0x1962ae00, /169.254.245.8:2738 => /169.254.245.8:5678] local: /169.254.245.8:5678 remote: /169.254.245.8:2738
10월 18, 2014 7:25:37 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:16656 remote: /169.254.245.8:2203 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:25:37 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:37 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:5678 remote: /169.254.245.8:2738 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-3-1413627932346"
state: FAILED
exit_code: 1
 
10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Force shutdown of: ProcessContainer{containedID='Node-1-1413627932150', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-1-1413627932150}
10월 18, 2014 7:25:37 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-1-1413627932150', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-1-1413627932150}
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0x823d655a, /169.254.245.8:2206 => /169.254.245.8:16656] local: /169.254.245.8:16656 remote: /169.254.245.8:2206
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:16656 remote: /169.254.245.8:2206 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0xd0d156ad, /169.254.245.8:2603 => /169.254.245.8:5678] local: /169.254.245.8:5678 remote: /169.254.245.8:2603
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:5678 remote: /169.254.245.8:2603 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-1-1413627932150"
state: FAILED
exit_code: 1
 
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager onEvaluatorException
경고: Failed evaluator: Node-3-1413627932346
com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-3-1413627932346] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onError
경고: Shutting down the Driver with an exception: 
java.lang.RuntimeException: No handler bound for FailedEvaluator: FailedEvaluator{id='Node-3-1413627932346'}
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-3-1413627932346] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	... 1 more

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'FAILING'
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager close
경고: Force resource release because the client closed the clock.
java.lang.IllegalStateException: Scheduling alarm on a closed clock
	at com.microsoft.wake.time.runtime.RuntimeClock.scheduleAlarm(RuntimeClock.java:78)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.close(EvaluatorManager.java:174)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onEvaluatorDone(EvaluatorManager.java:346)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onEvaluatorStatusMessage(EvaluatorManager.java:323)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onEvaluatorHeartbeatMessage(EvaluatorManager.java:299)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorHeartbeatHandler.onNext(EvaluatorHeartbeatHandler.java:58)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorHeartbeatHandler.onNext(EvaluatorHeartbeatHandler.java:33)
	at com.microsoft.wake.remote.impl.HandlerContainer.onNext(HandlerContainer.java:143)
	at com.microsoft.wake.remote.impl.HandlerContainer.onNext(HandlerContainer.java:34)
	at com.microsoft.wake.remote.impl.OrderedPullEventHandler.onNext(OrderedRemoteReceiverStage.java:167)
	at com.microsoft.wake.remote.impl.OrderedPullEventHandler.onNext(OrderedRemoteReceiverStage.java:149)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorCompletionHandler onNext
정보: Received CompletedEvaluator: CompletedEvaluator{id='Node-4-1413627932379'}
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Force shutdown of: ProcessContainer{containedID='Node-2-1413627932285', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-2-1413627932285}
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-2-1413627932285', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-2-1413627932285}
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0x52e43438, /169.254.245.8:2204 => /169.254.245.8:16656] local: /169.254.245.8:16656 remote: /169.254.245.8:2204
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.buffer.WrappedByteBuf.writeBytes(WrappedByteBuf.java:641)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:16656 remote: /169.254.245.8:2204 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0x4ebd2cef, /169.254.245.8:2604 => /169.254.245.8:5678] local: /169.254.245.8:5678 remote: /169.254.245.8:2604
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:5678 remote: /169.254.245.8:2604 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-2-1413627932285"
state: FAILED
exit_code: 1
 
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager onEvaluatorException
경고: Failed evaluator: Node-1-1413627932150
com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-1-1413627932150] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onError
경고: Shutting down the Driver with an exception: 
java.lang.RuntimeException: No handler bound for FailedEvaluator: FailedEvaluator{id='Node-1-1413627932150'}
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-1-1413627932150] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	... 1 more

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'FAILING'
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Force shutdown of: ProcessContainer{containedID='Node-4-1413627932379', nodeID='Node-4', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-4-1413627932379}
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-4-1413627932379', nodeID='Node-4', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-4-1413627932379}
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0xfd3802b4, /169.254.245.8:2605 => /169.254.245.8:5678] local: /169.254.245.8:5678 remote: /169.254.245.8:2605
10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:5678 remote: /169.254.245.8:2605 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-4-1413627932379"
state: FAILED
exit_code: 1
 
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager onEvaluatorException
경고: Failed evaluator: Node-2-1413627932285
com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-2-1413627932285] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onError
경고: Shutting down the Driver with an exception: 
java.lang.RuntimeException: No handler bound for FailedEvaluator: FailedEvaluator{id='Node-2-1413627932285'}
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-2-1413627932285] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	... 1 more

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'FAILING'
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Releasing Container with containerId [ProcessContainer{containedID='Node-2-1413627932285', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-2-1413627932285}]
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.Evaluators close
경고: Unclean shutdown of evaluator Node-3-1413627932346
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-2-1413627932285', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-2-1413627932285}
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.Evaluators close
경고: Unclean shutdown of evaluator Node-1-1413627932150
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.Evaluators close
경고: Unclean shutdown of evaluator Node-2-1413627932285
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.evaluator.Evaluators close
경고: Unclean shutdown of evaluator Node-4-1413627932379
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager sendJobEndingMessageToClient
경고: Sending message about an unclean driver shutdown.
java.lang.RuntimeException: No handler bound for FailedEvaluator: FailedEvaluator{id='Node-2-1413627932285'}
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-2-1413627932285] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	... 1 more

10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 3, Outstanding requests: 0
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Releasing Container with containerId [ProcessContainer{containedID='Node-1-1413627932150', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-1-1413627932150}]
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-1-1413627932150', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-1-1413627932150}
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 2, Outstanding requests: 0
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Releasing Container with containerId [ProcessContainer{containedID='Node-4-1413627932379', nodeID='Node-4', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-4-1413627932379}]
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-4-1413627932379', nodeID='Node-4', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-4-1413627932379}
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 1, Outstanding requests: 0
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Releasing Container with containerId [ProcessContainer{containedID='Node-3-1413627932346', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-3-1413627932346}]
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.driver.DriverRuntimeStopHandler onNext
정보: Driver shutdown complete
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-3-1413627932346', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:16656', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627930066\Node-3-1413627932346}
10월 18, 2014 7:25:38 오후 com.microsoft.reef.runtime.common.Launcher main
정보: Exiting Launcher.main()
